helm-chart/binderhub/templates/container-builder/daemonset.yaml

```diff
@@ -0,0 +1,125 @@
+{{- if ne .Values.imageBuilderType "local" -}}
+{{- $builderName := .Values.imageBuilderType }}
+{{- $builder := (index .Values $builderName) }}
+{{- $daemonset := $builder.daemonset }}
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Release.Name }}-{{ $builderName }}
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      name:  {{ .Release.Name }}-{{ $builderName }}
+  template:
+    metadata:
+      labels:
+        name: {{ .Release.Name }}-{{ $builderName }}
+        app: binder
+        component: image-builder
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+    spec:
+      {{- with include "jupyterhub.imagePullSecrets" (dict "root" . "image" $builder.daemonset.image) }}
+      imagePullSecrets: {{ . }}
+      {{- end }}
+      tolerations:
+      - effect: NoSchedule
+        key: hub.jupyter.org/dedicated
+        operator: Equal
+        value: user
+      - effect: NoSchedule
+        key: hub.jupyter.org_dedicated
+        operator: Equal
+        value: user
+      nodeSelector: {{ .Values.config.BinderHub.build_node_selector | default dict | toJson }}
+
+      {{- with $builder.initContainers }}
+      initContainers:
+        {{- . | toYaml | nindent 8 }}
+      {{- end }}
+
+      containers:
+        - name: {{ $builderName }}
+          image: {{ $daemonset.image.name }}:{{ $daemonset.image.tag }}
+          {{- with $daemonset.image.pullPolicy }}
+          imagePullPolicy: {{ . }}
+          {{- end }}
+          {{- with $daemonset.resources }}
+          resources:
+            {{- $daemonset.resources | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if eq $builderName "dind" }}
+          args:
+            - dockerd
+            - --storage-driver={{ .Values.dind.storageDriver }}
+            - -H unix://{{ .Values.dind.hostSocketDir }}/docker.sock
+            {{- with .Values.dind.daemonset.extraArgs }}
+            {{- . | toYaml | nindent 12 }}
+            {{- end }}
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - name: dockerlib-dind
+            mountPath: /var/lib/docker
+          - name: rundind
+            mountPath: {{ $builder.hostSocketDir }}
+          {{- end }}
+          {{- if eq $builderName "pink" }}
+          args:
+            - podman
+            - system
+            - service
+            - --time=0
+            - unix:///var/run/pink/docker.sock
```

Comment:
```suggestion
            - unix://{{ .Values.pink.hostSocketDir }}/docker.sock
```

Reply:
While it would make it on par with the `dind` implementation I have realized that it only works if `hostSocketDir` points to a folder that exists on both the host and pod filesystems. The default values do.

What made be aware of that issue is that I tested a deployment with the change just it case it would have some effect I did not consider. While implementing the Podman support, I have put the socket and storage in a dedicated folder structure to ensure I would not break the host system unintentionally. This folder is located in `/local`. I know this path is not part of the FHS but I am wondering whether we should take this case into consideration.

Reply:
`/var/lib/pink/` doesn't exist by default and it worked for me.

Reply:
I found the issue in my test !

The mountPath must also be changed to use `{{ .Values.pink.hostSocketDir }}` as it is done for the dind part. Otherwise, the parameter will not point to the correct location.